core/src/main/java/org/apache/iceberg/FastAppend.java

```diff
@@ -49,7 +49,8 @@
   private ManifestFile newManifest = null;
   private boolean hasNewFiles = false;
 
-  FastAppend(TableOperations ops) {
+  FastAppend(String tableName, TableOperations ops) {
+    //TODO we use tableName to publish notification
```

Comment:
I think the append event should be created in this PR if we are sending events from `MergingSnapshotProducer`. I don't think it is necessary to implement events for `SnapshotManager` yet, though.

Reply:
So `FastAppend` extends `SnapshotProducer` . If I change `SnapshotProducer`, I'll have to change `BaseRewriteManifests`. OK to make that change in this PR?

Reply:
Are you saying that `SnapshotProducer` would send the event? What needs to be changed?

Reply:
Scratch that. I think this can be supported. I'll do that as part of this pr